### PR TITLE
feat: support enums in composite types

### LINKED ISSUE #2605: Allow enums on MongoDB composites
The following is not possible at the moment for a MongoDB schema, but should be:
```prisma
model Test {
    id Int @id @map("_id")
    test TestComposite
}

type TestComposite {
    field TestEnum
}

enum TestEnum {
    Foo,
    Bar
}
```

### PR DESCRIPTION
**Pull request by Weakky**
## Overview

Closes #2605

### CODE REVIEW

File: libs/datamodel/core/tests/attributes/default_negative.rs [hunk 1]
```diff
@@ -186,7 +186,7 @@ fn must_error_if_default_value_for_enum_is_not_valid() {
     let error = datamodel::parse_schema(dml).map(drop).unwrap_err();
 
     let expectation = expect![[r#"
-        [1;91merror[0m: [1mError parsing attribute "@default": The defined default value is not a valid value of the enum specified for the field.[0m
+        [1;91merror[0m: [1mError parsing attribute "@default": The defined default value `B` is not a valid value of the enum specified for the field.[0m
```
Comment (tomhoule) [inline libs/datamodel/core/tests/attributes/default_negative.rs:189]: :+1:

**Review by tomhoule — APPROVED** [review]
looks good to me :+1:

File: query-engine/prisma-models/src/extensions.rs [hunk 2]
```diff
@@ -136,3 +131,42 @@ impl DatamodelFieldExtensions for dml::ScalarField {
         }
     }
 }
+
+impl DatamodelFieldExtensions for dml::CompositeTypeField {
+    fn type_identifier(&self) -> TypeIdentifier {
+        match &self.r#type {
+            datamodel::CompositeTypeFieldType::CompositeType(_) => {
+                todo!("composite type support in datamodel_converter")
```
Comment (dpetrick) [inline query-engine/prisma-models/src/extensions.rs:139]: This isn't really a todo, this is a an unreachable in my opinion.

**Review by dpetrick — COMMENTED** [review]
Okay from the QE side.

**Pull request merged**